### PR TITLE
add new telemetry events for features behind a SKU

### PR DESCRIPTION
**Pull request by Willyfrog**
#### Summary

Added events for telemetry: 
- guests
- custom groups
- shared channels
- read-only channels

#### Ticket Link
[MM-60580](https://mattermost.atlassian.net/browse/MM-60580)

[MM-60580]: https://mattermost.atlassian.net/browse/MM-60580?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: source/manage/telemetry.rst [hunk 1]
```diff
@@ -161,6 +161,10 @@ Non-personally Identifiable Diagnostic Information, distinguished by end users a
   - *System Console Menu Discovery Diagnostics:* Clicks on the hamburger menu items of the System Console, including Administrator's Guide, Troubleshooting Forum, Commercial Support, About Mattermost, and clicks on the left-hand side navigation menu items
   - *In Product Notices Diagnostics:* Notices viewed, and the notices on which an action button was clicked.
   - *Threaded discussions:* Clicks to reply to a thread, reply using the footer element, filter threads by unread, mark as read, access to global threads section.
+  - *Custom Groups:* Invite people to a channel by using a custom group, mention a custom group, or modify a custom group.
+  - *Read-Only Channels:* Navigate to a read-only channel, post a message to a read-only channel.
```
Comment (cwarnermm) [inline source/manage/telemetry.rst:165]:
```suggestion
  - *Read-Only Channels:* Navigate to a read-only channel, and post a message to a read-only channel.
```

File: source/manage/telemetry.rst [hunk 2]
```diff
@@ -161,6 +161,10 @@ Non-personally Identifiable Diagnostic Information, distinguished by end users a
   - *System Console Menu Discovery Diagnostics:* Clicks on the hamburger menu items of the System Console, including Administrator's Guide, Troubleshooting Forum, Commercial Support, About Mattermost, and clicks on the left-hand side navigation menu items
   - *In Product Notices Diagnostics:* Notices viewed, and the notices on which an action button was clicked.
   - *Threaded discussions:* Clicks to reply to a thread, reply using the footer element, filter threads by unread, mark as read, access to global threads section.
+  - *Custom Groups:* Invite people to a channel by using a custom group, mention a custom group, or modify a custom group.
```
Comment (cwarnermm) [inline source/manage/telemetry.rst:164]:
```suggestion
  - *Custom Groups:* Invite people to a channel by using a custom group, mention a custom group, and modify a custom group.
```

File: source/manage/telemetry.rst [hunk 3]
```diff
@@ -161,6 +161,10 @@ Non-personally Identifiable Diagnostic Information, distinguished by end users a
   - *System Console Menu Discovery Diagnostics:* Clicks on the hamburger menu items of the System Console, including Administrator's Guide, Troubleshooting Forum, Commercial Support, About Mattermost, and clicks on the left-hand side navigation menu items
   - *In Product Notices Diagnostics:* Notices viewed, and the notices on which an action button was clicked.
   - *Threaded discussions:* Clicks to reply to a thread, reply using the footer element, filter threads by unread, mark as read, access to global threads section.
+  - *Custom Groups:* Invite people to a channel by using a custom group, mention a custom group, and modify a custom group.
+  - *Read-Only Channels:* Navigate to a read-only channel, and post a message to a read-only channel.
+  - *Shared Workspaces:* Navigate to a shared channel, post a message in a shared channel, and mention a remote user.
+  - *Guest Accounts:* Mention a guest account, DM a guest, add a guest to a channel.
```
Comment (cwarnermm) [inline source/manage/telemetry.rst:167]:
```suggestion
  - *Guest Accounts:* Mention a guest account, directly message a guest, and add a guest to a channel.
```

**Review by cwarnermm — APPROVED** [review]
Thanks, @Willyfrog!!

**Comment by cwarnermm** [comment]
@Willyfrog - Can this docs PR be merged and published now, or is this update tied to Engineering product updates?

**Comment by Willyfrog** [comment]
> @Willyfrog - Can this docs PR be merged and published now, or is this update tied to Engineering product updates?

There is a PR waiting to be merged. Once that's done, I'm not sure if we should wait for the version to come out or if we could publish right away (since it would be available on a dev environment)

PR: https://github.com/mattermost/mattermost/pull/29024

**Comment by Willyfrog** [comment]
@cwarnermm the other PR is merged, not sure if we should merge this one or wait until release.

**Comment by cwarnermm** [comment]
Thanks, @Willyfrog. That Engineering PR doesn't yet have a milestone on it. As soon as it gets into a release queue, I'll update the milestone on this docs PR to match.

**Pull request merged**